pypga/modules/migen/spimachine.py

```diff
@@ -72,6 +73,7 @@ def __init__(self, width):
                 Cat(sr[1:], self.sdi),
                 Cat(self.sdi, sr[:-1])))
```

Comment:
This basically prepends or appends the current bit to `sdi`, right?

Reply:
this sets `pdi` to `[sdi, sr[0], sr[1], ..., sr[-2]]` (lsb_first=0) or `[sr[1], sr[2], ..., sr[-1], sdi]` (lsb_first=1). `sr` stands for "shift register", so every SPI cycle its bits are shifted left/right by one.